config/initializers/unify_google_compute.rb

```diff
@@ -100,15 +150,31 @@ def create_disk(params)
             # rethought.
             size_gb: 10,
             zone_name: params[:zone],
-            source_image: params[:image_id],
             autoDelete: true
           }
+
+          id = real_id(params[:image_id])
+          if snapshot?(params[:image_id])
+            disk_params[:sourceSnapshot] = id
+          else
+            disk_params[:source_image] = id
+          end
+
           disk = service.disks.create(disk_params)
           disk.wait_for { disk.ready? }
 
           disk
         end
 
+        def snapshot?(id_at_site)
+          id_at_site.start_with?('snapshot:')
+        end
+
+        def real_id(id_at_site)
+          match = id_at_site.match(/\A.*:(.*)\z/)
+          match ? match[1] : id_at_site
+        end
```

Comment:
There are 2  identical definitions of snapshot? and real_id methods. Could you define it in one place and reuse it?